src/control/lib/control/pool.go

```diff
@@ -1227,105 +1227,139 @@ func ListPools(ctx context.Context, rpcClient UnaryInvoker, req *ListPoolsReq) (
 	return resp, nil
 }
 
-// Return the maximal SCM and NVMe size of a pool which could be created with all the storage nodes.
-func GetMaxPoolSize(ctx context.Context, log logging.Logger, rpcClient UnaryInvoker, ranks []system.Rank) (uint64, uint64, error) {
-	storageScanReq := &StorageScanReq{Usage: true}
-	resp, err := StorageScan(ctx, rpcClient, storageScanReq)
-	if err != nil {
-		return 0, 0, err
-	}
+type rankFreeSpaceMap map[system.Rank]uint64
 
-	if len(resp.HostStorage) == 0 {
-		return 0, 0, errors.New("No DAOS server available")
+type checkRankFn func(rank system.Rank) bool
+
+func newCheckRankFunc(ranks []system.Rank) checkRankFn {
+	return func(rank system.Rank) bool {
+		return len(ranks) == 0 || rank.InList(ranks)
 	}
+}
 
-	hostStorageMap := resp.HostStorage
-	// DAOS-10885: The variable ranksWithoutSmd is used as a marking board for checking if there
-	// are any ranks with SCM storage but no NVMe.
-	ranksWithoutSmd := make(map[system.Rank]*storage.ScmNamespace)
-	var scmBytes uint64 = math.MaxUint64
-	var nvmeBytes uint64 = math.MaxUint64
-	for _, key := range hostStorageMap.Keys() {
-		hostStorageSet := hostStorageMap[key]
-		hostStorage := hostStorageSet.HostStorage
+// Add namespace ranks to rankNVMeFreeSpace map and return minimum free available SCM namespace bytes.
+func processSCMSpaceStats(log logging.Logger, checkRank checkRankFn, scmNamespaces storage.ScmNamespaces, rankNVMeFreeSpace rankFreeSpaceMap) (uint64, error) {
+	scmBytes := uint64(math.MaxUint64)
 
-		if hostStorage.ScmNamespaces.Free() == uint64(0) {
-			return 0, 0, errors.Errorf("Host without SCM storage: hostname=%s",
-				hostStorageSet.HostSet.String())
+	for _, scmNamespace := range scmNamespaces {
+		if scmNamespace.Mount == nil {
+			// FIXME: Under what circumstances is this valid, should this
```

Comment:
From my side, not enough knowledge of the DAOS internals to define if such situation is managed or if an error should be raised.

Reply:
I think we should make this an error, if not it will give us obscure behaviour that may be hard to diagnose

Reply:
OK I will do it.